### DISCUSSION TITLE
What auto-generated Serenity files need checked into source control?

### DISCUSSION THREAD

**Discussion by rymk14** [serenity-is/Serenity#7239] [Help] [answered]
Hi - I was wondering what Serenity files you all add to your .gitignore?

We are using Serenity for an admin app and have been checking all of the `wwwroot/esm/*` files and `/Modules/ServerTypes/*` files into git.

As the project has grown, we now encounter a lot of merge conflicts on the Serenity auto-generated files and we trying to figure out what files we can stop checking into source control. We would like to stop checking in files that can be rebuilt later.

Can we stop checking in all of `wwwroot/*` into git? Or at least stop checking in 'wwwroot/esm/_chunks/*` as we get a lot of conflicts here.

Can we stop checking in `Modules/ServerTypes`? In the past, we think we had an issue wtih these files being excluding and not working on other devs machines.

Thanks for your feedback!

**Comment by VictorTomaili** [comment, marked as answer]
If you build and regenerate esm files before deployment, they can be excluded. 
Server types are generated when the related files are changed. So you need to keep them in source to avoid build errors and avoid double build etc.

**Reply by rymk14** [reply]
awesome, thank you for the explanation.
